webserver/webserver/src/main/java/io/helidon/webserver/Request.java

```diff
@@ -218,41 +219,41 @@ private UriInfo createRequestedUri() {
         String path = null;
         String query = query();
 
-        boolean discovered = false;
-        for (var type : bareRequest.socketConfiguration().requestedUriDiscoveryTypes()) {
-            switch (type) {
-            case FORWARDED -> {
-                List<Forwarded> forwardedList = Forwarded.create(headers);
-                if (!forwardedList.isEmpty()) {
-                    Forwarded f = forwardedList.get(0);
-                    scheme = f.proto().orElse(null);
-                    authority = f.host().orElse(null);
-                    discovered = true;
+        // Trusted proxies could be null if the discovery type is Host, either explicitly or by default.
+        AllowList trustedProxies = bareRequest.socketConfiguration().trustedProxies();
+        if (trustedProxies != null && trustedProxies.test(hostPart(remoteAddress()))) {
```

Comment:
The null check should be removed. The `trustedProxies` cannot be null.

Reply:
fixed